### DISCUSSION TITLE
Default memory allocated to Lua VM

### DISCUSSION THREAD

**Discussion by kollol-chowdhury-incontact** [Kong/kong#8870] [Help]
Hi Team,

Do we know how much memory is allocated to Lua VM as i am getting nginx: lua atpanic: Lua VM crashed, reason: not enough memory.
Kong version:2.7.1
Instance type:c5.2xlarge

TIA

**Comment by jschmid1** [comment]
Based on your instance type you should have enough memory available. Do you have any other relevant processes running on this machine?

The memory consumption is dependent on multiple factors but you can expect a couple hundred megabytes of memory for a kong instance.

**Comment by kollol-chowdhury-incontact** [comment]
We are running it in EC2 (c5.2x large)
And we are running kong in docker container.
No other process is running. Out of 16GB RAM, just 4GB is used by OS and other processes.

Do we have any option to check the memory allocated to VM because we are getting Out of memory exception .

**Reply by hanshuebner** [reply]
Hello @kollol-chowdhury-incontact,

you may need to tune the memory allocation parameters of your Docker runtime.  The machine itself should have enough memory to accommodate a rather large installation of Kong.  Use the Linux diagnostics tools (ps, top, htop etc.) to diagnose the memory consumption of the processes running on the Docker host system.

Kind regards,
Hans
